microprofile/opentracing-extension/src/main/java/org/wildfly/extension/microprofile/opentracing/TracingDeploymentProcessor.java

```diff
@@ -210,6 +211,7 @@ private void injectTracer(DeploymentPhaseContext deploymentPhaseContext, Capabil
                 tracer = WildFlyTracerFactory.getTracer(tracerConfigurationName, serviceName);
             }
         }
+        TracingCDIExtension.registerApplicationTracer(moduleCL, tracer);
         deploymentUnit.addToAttachmentList(ServletContextAttribute.ATTACHMENT_KEY, new ServletContextAttribute(SMALLRYE_OPENTRACING_SERVICE_NAME, serviceName));
         deploymentUnit.addToAttachmentList(ServletContextAttribute.ATTACHMENT_KEY, new ServletContextAttribute(SMALLRYE_OPENTRACING_TRACER, tracer));
```

Comment:
You could probably remove this now, the only usage is in `TracerDynamicFeature` which again needs req. context active (not sure if it can be a problem there). Instead, you could either `@Inject` the `Tracer` if that's supported (not sure now, I might just be confused from Quarkus) or use `CDI.current()` to grab it.

Reply:
It broke a lot of tests related to CDI